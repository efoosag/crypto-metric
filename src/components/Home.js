/* eslint-disable react/jsx-indent */
/* eslint-disable indent */
/* eslint-disable function-paren-newline */
/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable comma-dangle */
import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import CoinItem from './CoinItem';
import { getCoinData } from '../redux/fetchApi';

const Home = () => {
  const dispatch = useDispatch();
  const cryptos = useSelector((state) => state.CoinReducer);

  useEffect(() => {
    if (cryptos) {
      dispatch(getCoinData());
    }
  }, []);

  const [value, setValue] = useState('');
  const searchHandler = (e) => {
    setValue(e.target.value);
  };

  return (
    <div>
      <form>
        <br />
        <input
          placeholder="SEARCH BY CRYPTO COIN"
          className="search"
          type="text"
          value={value}
          onChange={searchHandler}
        />
      </form>
      <div>
        {' '}
        {value
          ? cryptos.coin
              .filter((coinData) =>
                coinData[1].name
                  .toLowerCase()
                  .includes(value.toLocaleLowerCase())
              )
              .map((coin) => <CoinItem key={coin[0]} crypto={coin[1]} />)
          : cryptos.coin.map((coin) => (
              <CoinItem key={coin[0]} crypto={coin[1]} />
            ))}
      </div>
    </div>
  );
};

export default Home;
