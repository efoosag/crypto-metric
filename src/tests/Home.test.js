/* eslint-disable comma-dangle */
/* eslint-disable import/no-named-as-default */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../components/Home';

describe('Home Component test', () => {
  it('Home Page', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
