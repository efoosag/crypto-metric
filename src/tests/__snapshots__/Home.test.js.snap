// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component test Home Page 1`] = `
<div>
  <form
    className="search"
  >
    <br />
    <input
      className="search"
      onChange={[Function]}
      placeholder="SEARCH CRYPTO COIN"
      type="text"
      value=""
    />
  </form>
  <div
    className="cryptoCase"
  >
     
  </div>
</div>
`;
